src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -223,5 +223,378 @@ function (indicator_max::IndicatorMax)(u::AbstractArray{<:Any,4},
   return alpha
 end
 
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+function create_cache(::Type{IndicatorANN}, equations::AbstractEquations{2}, basis::LobattoLegendreBasis, indicator_type)
+
+  alpha = Vector{real(basis)}()
+  alpha_tmp = similar(alpha)
+  A = Array{real(basis), ndims(equations)}
+
+  if indicator_type == "NNPP"
+    indicator_threaded  = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+    modal_threaded      = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+    modal_tmp1_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+
+    return (; alpha, alpha_tmp, indicator_threaded, modal_threaded, modal_tmp1_threaded)
+  elseif indicator_type == "NNRH"
+    indicator_threaded  = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+    modal_threaded      = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+    modal_tmp1_threaded = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+    #X = Vector{Float64}(undef, 3, nelements(dg, cache))
+    #network_input = Vector{Float64}(undef, 15)
+
+    return (; alpha, alpha_tmp, indicator_threaded, modal_threaded, modal_tmp1_threaded) #, X, network_input)
+
+  elseif indicator_type == "CNN"
+    indicator_threaded  = [A(undef, nnodes(basis), nnodes(basis)) for _ in 1:Threads.nthreads()]
+
+    return (; alpha, alpha_tmp, indicator_threaded)
+  end
+end
+
+# this method is used when the indicator is constructed as for AMR
+function create_cache(typ::Type{IndicatorANN}, mesh, equations::AbstractEquations{2}, dg::DGSEM, cache, indicator_type)
+  create_cache(typ, equations, dg.basis, indicator_type)
+end
+
+
+function (indicator_ann::IndicatorANN)(u::AbstractArray{<:Any,4}, mesh,
+                                         equations, dg::DGSEM, cache;
+                                         kwargs...)
```

Comment:
```suggestion
function (indicator_ann::IndicatorANN)(u, mesh::TreeMesh{2},
                                      equations, dg::DGSEM, cache;
                                      kwargs...)
```
I think this indicator works only for the `TreeMesh` anyways